src/core/layout/qgslayoutitemattributetable.cpp

```diff
@@ -799,7 +799,9 @@ bool QgsLayoutItemAttributeTable::readPropertiesFromElement( const QDomElement &
   mVectorLayer.resolveWeakly( mLayout->project() );
 
   //connect to new layer
-  connect( sourceLayer(), &QgsVectorLayer::layerModified, this, &QgsLayoutTable::refreshAttributes );
+  QgsVectorLayer *newLayer = sourceLayer();
```

Comment:
Minor thing, but in cases like this I think it's nicer to condense into a single line so that the variable has a tight scope:

```
if ( QgsVectorLayer* new layer = ... )
```